patches/0007-Add-backend-code-gen.patch

```diff
@@ -0,0 +1,427 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Davis Goodin <dagood@microsoft.com>
+Date: Fri, 23 Jun 2023 11:58:31 -0500
+Subject: [PATCH] Add backend code gen
+
+Adds code generation that works with the final list of crypto backends
+to add extra features and tie up loose ends in an easily maintained way.
+
+Backend conflict error generator: creates files that generate
+informative build errors when the backends aren't configured correctly.
+
+"nobackend" build constraint generator: gathers the build constraints
+for all the backends to create the "nobackend" constraint.
+
+The generators also run during standard test runs in validation mode to
+ensure the generated content is not out of date when backends are
+added/removed or when their build constraints change.
+
+To simplify patch maintenance, the generators are only run once, in this
+patch. An alternative would be adding the generators in an earlier patch
+and re-running the generators immediately for each backend patch. This
+would make the direct connection between changes clearer, but make it
+harder to update the generators and to deal with conflicts.
```

Comment:
I did write the generator so it could work if run per-patch (that's what `go1.1` is about), and actually tried it out. I think it's significantly easier to maintain in its own patch, though. Needing to rebase, run a command on each backend patch, amend, and continue is not great with our current tools, IMO.

If we added a "patch fixup command" patch file feature that would run in a "patch fixup rebase" `git-go-patch` command, that would potentially make it viable. I've always thought this kind of feature would be useful for re-vendoring, too, but it just isn't quite useful enough to be worthwhile vs. simply having the command in the patch message. (Or remembering it--vendoring is pretty straightforward.)

(If you're in the middle of a backend patch, you also need to run with `go generate -mod=readonly ...` because the vendoring dir is stale at that point in time.)